core/src/wallet/cosmos/CosmosLikeAccount.cpp

```diff
@@ -82,41 +82,123 @@ namespace ledger {
+                               out.amount = sent_amount - recv_amount;
+                               out.type = api::OperationType::SEND;
+                        }
+
+                    } break;
+                    case api::CosmosLikeMsgType::MSGDELEGATE: {
+                        out.amount = boost::get<cosmos::MsgDelegate>(msg.content).amount.amount;
+                        out.type = api::OperationType::NONE;
+                    } break;
+                    case api::CosmosLikeMsgType::MSGUNDELEGATE: {
+                        out.amount = boost::get<cosmos::MsgUndelegate>(msg.content).amount.amount;
+                        out.type = api::OperationType::NONE;
+                    } break;
+                    case api::CosmosLikeMsgType::MSGBEGINREDELEGATE: {
+                        out.amount =
+                            boost::get<cosmos::MsgBeginRedelegate>(msg.content).amount.amount;
+                        out.type = api::OperationType::NONE;
+                    } break;
+                    case api::CosmosLikeMsgType::MSGSUBMITPROPOSAL: {
+                        const auto &coins =
+                            boost::get<cosmos::MsgSubmitProposal>(msg.content).initialDeposit;
+                        std::for_each(coins.begin(), coins.end(), [&](const cosmos::Coin& amount) {
+                            out.amount = out.amount + BigInt::fromDecimal(amount.amount);
+                        });
+                        const auto &sender =
+                            boost::get<cosmos::MsgSubmitProposal>(msg.content).proposer;
+                        if (sender == address) {
+                            out.type = api::OperationType::SEND;
                         }
+                        out.type = api::OperationType::NONE;
+                    } break;
+                    case api::CosmosLikeMsgType::MSGDEPOSIT: {
+                        const auto &coins = boost::get<cosmos::MsgDeposit>(msg.content).amount;
+                        std::for_each(coins.begin(), coins.end(), [&](cosmos::Coin amount) {
+                            out.amount = out.amount + BigInt::fromDecimal(amount.amount);
+                        });
+                        const auto &sender = boost::get<cosmos::MsgDeposit>(msg.content).depositor;
+                        if (sender == address) {
+                            out.type = api::OperationType::SEND;
+                        }
+                        out.type = api::OperationType::NONE;
+                    } break;
+                    case api::CosmosLikeMsgType::MSGVOTE:
+                    case api::CosmosLikeMsgType::MSGWITHDRAWDELEGATIONREWARD:
+                    case api::CosmosLikeMsgType::MSGCREATEVALIDATOR:
+                    case api::CosmosLikeMsgType::MSGEDITVALIDATOR:
+                    case api::CosmosLikeMsgType::MSGSETWITHDRAWADDRESS:
+                    case api::CosmosLikeMsgType::MSGWITHDRAWDELEGATORREWARD:
```

Comment:
Ain't this a RECEIVE? Just without known amount...

Reply:
There's no amount in the operation anyway, so it's better to display nothing than "RECEIVE 0 ATOM" for now.

Catching the real amount means better parsing of events within a block/tx/message and we don't do it yet. I'll probably create a ticket for that